omod/src/main/java/org/openmrs/module/ssemrws/web/controller/SSEMRWebServicesController.java

```diff
@@ -564,6 +566,33 @@ private static boolean determineIfPatientIsNewClient(Patient patient, Date start
 		// return false;
 	}
 	
+	private static String getDateOfLastRefill(Date startDate, Date endDate) {
```

Comment:
@Michaelndula this is a highly inefficient approach that may still not work as expected. Try this out:

```	private static String getLastRefillDate(Patient patient, Date endDate) {
		Concept lastRefillDateConcept = Context.getConceptService().getConceptByUuid(LAST_REFILL_DATE_CONCEPT_UUID);
		List<Obs> LastRefillDateObs = Context.getObsService().getObservations(Collections.singletonList(patient.getPerson()), null, Collections.singletonList(lastRefillDateConcept), null, null, null, null, 1, null, null, endDate, false);
		
		String lastRefillDate = "";
		if (LastRefillDateObs != null && !LastRefillDateObs.isEmpty()) {
			dateTimeFormatter.format(LastRefillDateObs.get(0).getValueDate());
		}
		return lastRefillDate;
	}

```

Reply:
@moshonk thank you for pointing this out, I have resolved it.